### PR TITLE
333 configure step count in dialogs

### PR DESCRIPTION
**Pull request by wbaccinelli**
Added intensity minutes.

@bscheltinga, please check from commit 4aeb9faacfb9e9f8a55e0cec6c061e649761dd55

Also fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/387
and fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/334
and fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/339
and even fixes https://github.com/PerfectFit-project/virtual-coach-issues/issues/416

### CODE REVIEW

File: scheduler/requirements.txt [hunk 1]
```diff
@@ -4,6 +4,6 @@ numpy
 python-dotenv
 git+https://github.com/PerfectFit-project/virtual-coach-db@v0.1.24
 git+https://github.com/PerfectFit-project/niceday_client@v0.1.5
-git+https://github.com/PerfectFit-project/virtual-coach-rasa@main#subdirectory=sensor_api
+git+https://github.com/PerfectFit-project/virtual-coach-rasa@333-configure-step-count-in-dialogs#subdirectory=sensor_api
```
Comment (bscheltinga) [inline scheduler/requirements.txt:7]: We should change this before merging right?

**Review by bscheltinga — APPROVED** [review]
Hi Walter, nice! Looks good :)

File: Rasa_Bot/actions/actions_weekly_reflection.py [hunk 2]
```diff
@@ -279,35 +297,84 @@ def name(self):
         return "action_set_intensity_minutes"
 
     async def run(self, dispatcher, tracker, domain):
-        ## TODO This method should get minutes and set the slot accordingly
-        intensity_minutes = 50
+
+        user_id = int(tracker.current_state()['sender_id'])
+        end = datetime.datetime.now()
+        start = end - datetime.timedelta(days=16)
```
[the remark this answers is not in the document]
Comment (wbaccinelli) [reply Rasa_Bot/actions/actions_weekly_reflection.py:303]: I think that I used the same number of days that was used `action_set_step_goal_days`. Is it correct that they are different?

File: Rasa_Bot/actions/actions_weekly_reflection.py [hunk 3]
```diff
@@ -279,35 +297,84 @@ def name(self):
...
+        previous_goal = get_intensity_minutes_goal(user_id)
+
+        # the first time, it's set to 15 minutes  ## Bouke: value from last week instead.
+        if previous_goal is None:
+            end = datetime.datetime.now()
+            start = end - datetime.timedelta(days=7)
+
+            previous_goal = get_intensity_minutes_data(user_id, start, end)
+            if previous_goal is None or previous_goal < 10:
+                previous_goal = 10
```
[the remark this answers is not in the document]
Comment (wbaccinelli) [reply Rasa_Bot/actions/actions_weekly_reflection.py:337]: I agree

File: scheduler/requirements.txt (hunk 1 above)
Comment (wbaccinelli) [reply scheduler/requirements.txt:7]: Now this subdirectory does not exist in the main, so we'd probably better do that after

File: Rasa_Bot/actions/actions_weekly_reflection.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bscheltinga) [reply Rasa_Bot/actions/actions_weekly_reflection.py:304]: It is slightly different, also I realize something else... The function get_intensity_minutes_data returns the total number of intensity minutes over the specified period. As we are setting a goal for 7 days we should take the intensity minutes over 7 days.
I also think that rounding would be nice. I did suggestions to round it to the nearest ten-fold

Comment (bscheltinga) [inline Rasa_Bot/actions/actions_weekly_reflection.py:304]:
```suggestion
        start = end - datetime.timedelta(days=7)

```

File: Rasa_Bot/actions/actions_weekly_reflection.py [hunk 4]
```diff
@@ -279,35 +297,84 @@ def name(self):
...
     async def run(self, dispatcher, tracker, domain):
-        ## TODO This method should get minutes and set the slot accordingly
-        intensity_minutes = 50
+
+        user_id = int(tracker.current_state()['sender_id'])
+        end = datetime.datetime.now()
+        start = end - datetime.timedelta(days=16)
+
+        intensity_minutes = get_intensity_minutes_data(user_id, start, end)
+
```
Comment (bscheltinga) [inline Rasa_Bot/actions/actions_weekly_reflection.py:307]:
```suggestion
        intensity_minutes = get_intensity_minutes_data(user_id, start, end)
        intensity_minutes = round(intensity_minutes, -1)


```

**Comment by bscheltinga** [comment]
Before merging, I'm currently working on a branch of this branch, to fix: https://github.com/PerfectFit-project/virtual-coach-issues/issues/381

**Comment by wbaccinelli** [comment]
The branch is included in [sensors-testing](https://github.com/PerfectFit-project/virtual-coach-rasa/tree/sensors-testing) branch

**Pull request closed without merging**